doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,773 @@
+```
+$ curl -i localhost:5588
+   HTTP/1.1 200 OK
+   Content-Length: 0
+   Server: Jetty(8.y.z-SNAPSHOT)
+```
+
+Health check service response when the service `xroad-signer` is not running.
+```
+$ curl -i localhost:5588
+HTTP/1.1 500 Server Error
+Transfer-Encoding: chunked
+Server: Jetty(8.y.z-SNAPSHOT)
+
+Fetching health check response timed out for: Authentication key OCSP status
+```
+
+
+## 4. Database replication setup
+
+For technical details on the PostgreSQL replication, refer to the [official documentation](https://www.postgresql.org/docs/9.2/static/high-availability.html).
+Note that the versions of PostgreSQL distributed with RHEL and Ubuntu are different. At the time of writing, RHEL 7
+distributes PostgreSQL version 9.2 and Ubuntu 14.04 distributes version 9.3; the replication configuration is the same
+for both versions.
+
+
+### 4.1 Setting up TLS certificates for database authentication
+This section describes how to create and set up certificate authentication between the slave and master database instances.
+
+For further details on the certificate authentication, see the
+[PostgreSQL documentation](https://www.postgresql.org/docs/9.2/static/auth-methods.html#AUTH-CERT).
+
+
+1. Generate the Certificate Authority key and a self-signed certificate for the root-of-trust:
+
+   ```
+   openssl req -new -x509 -days 7300 -nodes -sha256 -out master.crt -keyout master.key -subj '/O=cluster/CN=master'
+   ```
+   The subject name does not really matter here. Remember to keep the `master.key` file in a safe place.
+
+2. Generate keys and certificates signed by the CA for each postgresql instance, including the master. Do not use the CA
+   certificate and key as the database certificate and key.
+
+   ```
+   openssl req -new -nodes -days 7300 -keyout server.key -out server.csr -subj "/O=cluster/CN=<nodename>"
+   ```
+
+   **Note:** The `<nodename>` must match a replication user name; otherwise the subject name does not matter
+
+   ```
+   openssl x509 -req -in server.csr -CAcreateserial -CA master.crt -CAkey master.key -days 7300 -out server.crt
+   ```
+
+3. Copy the certificates to `/etc/xroad/postgresql` on each cluster instance:
+
+   ```bash
+   sudo mkdir -p -m 0755 /etc/xroad/postgresql
+   sudo chmod o+x /etc/xroad
+   ```
+   Copy the certificates, then copy one server key per instance so that each instance has a unique key:
+
+   ```bash
+   sudo chown postgres /etc/xroad/postgresql/server.key
+   sudo chmod 400 /etc/xroad/postgresql/server.key
+   ```
+
+> Alternatively, an existing internal CA can be used for managing the certificates. A sub-CA should be created as the
+> database cluster root-of-trust and use that for issuing the slave and master certificates.
+
+
+### 4.2 Creating a separate PostgreSQL instance for the `serverconf` database
+
+#### 4.2.1 on RHEL
+
+Create a new `systemctl` service unit for the new database. As root, execute the following scripts:
```

Comment:
(as there's a separate introduction to the next "Create the database" section, so I am getting the feeling this part does not talk about that...)